input-profile.ts

```diff
@@ -0,0 +1,360 @@
+import {HandTracking} from './hand-tracking.js';
+import {vec3, quat} from 'gl-matrix';
+import {property} from '@wonderlandengine/api/decorators.js';
+import {VrModeActiveSwitch} from './vr-mode-active-switch.js';
+
+const _tempVec = vec3.create();
+const _tempQuat = quat.create();
+const _tempRotation1 = new Float32Array(4);
+const _tempRotation2 = new Float32Array(4);
+const minTemp = new Float32Array(3);
+const maxTemp = new Float32Array(3);
+const hands = ['left', 'right'];
+
+interface VisualResponse {
+    target: Object3D;
+    min: Object3D;
+    max: Object3D;
+    id: number;
+}
+
+/**
+ * Dynamically load and map input profiles for XR controllers
+ * @extends {Component}
+ */
+
+export class InputProfile extends Component {
+    static TypeName = 'input-profile';
+    /**
+     * A cache to store loaded profiles for reuse.
+     * @type {Map<string, any>}
+     */
+    static Cache: Map<string, any> = new Map();
+    private _gamepadObjects: Record<string, Object3D> = {};
+    private _controllerModel: Object3D | null = null;
+    private _defaultControllerComponents: Component[] | undefined;
+    private _handedness!: string;
+    private _profileJSON: any = null;
+    private _buttons: VisualResponse[] = [];
+    private _axes: VisualResponse[] = [];
+
+    /**
+     * The XR gamepad associated with the current input source.
+     */
+    gamepad: Gamepad | undefined;
+    /**
+     * A Reference to the emitter which triggered on model lodaed event
+     * @type {Emitter}
+     */
+    onModelLoaded: Emitter = new Emitter();
+
+    /** returns url of input profile json file
+     *  @type {string}
+     */
+    url!: string;
+
+    /**
+     * A set of components to filter during component retrieval.
+     * @type {Set<string>}
+     */
+    toFilter: Set<string> = new Set(['vr-mode-active-mode-switch']);
+
+    @property.object()
+    defaultController!: Object3D;
+
+    @property.enum(hands, 0)
+    handednessIndex: number = 0;
+
+    @property.string(
+        'https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@latest/dist/profiles/'
+    )
+    path!: string;
+
+    @property.string()
+    customProfileFolder!: string;
+
+    @property.bool(false)
+    mapToDefaultController!: boolean;
+
+    @property.object()
+    trackedHand!: Object3D;
+
+    @property.bool(true)
+    addVrModeSwitch!: boolean;
+
+    start() {
+        this._controllerModel = null;
+        this.toFilter = new Set(['vr-mode-active-mode-switch']);
+        this._defaultControllerComponents = this._getComponents(this.defaultController);
+        this._handedness = hands[this.handednessIndex];
+
+        this.engine.onXRSessionStart.add(() => {
+            this.engine.xr?.session.addEventListener(
+                'inputsourceschange',
+                this._onInputSourcesChange.bind(this)
+            );
+        });
+    }
+
+    /**
+     * Cleans up resources when the component is deactivated.
+     */
+    onDeactivate() {
```

Comment:
Since `onDeactivate` is inherited from `Component`, here you can simply do nothing